<a href="https://colab.research.google.com/github/john-hafner/MSU-PHY/blob/main/PHY493_HW2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# PHY 492 HW 2
## John Hafner

In [ ]:
import numpy as np
import sympy as sym

import matplotlib.pyplot as plt

# \#1 (5+5+10+10 pts)
Consider the particle described by the 4-vector: $p_\mu = (200, 30, 100, 150)$
GeV in the lab frame.

- (a) What is the mass of this particle?  Which elementary particle is it?

Using the invariant relation:

$$E^2 - \vec{p}^2 = m^2c^4$$

We can solve for $m$:

$$m = c^2\sqrt{E^2 - p^2}$$

We have $E = 200$ GeV

and $\vec{p} = (30, 100, 150)$

So $\vec{p}^2 = 30^2 + 100^2 + 150^2 =$ (see below)

In [42]:
c = 3e8
p4_w = np.array([[200, 30, 100, 150]]).T # declaring a 4-vector
m = float(np.sqrt(p4_w[0]**2 - float(p4_w[1:].T@p4[1:])))
print('mass = %.2f GeV/c^2' % (m))

mass = 81.24 GeV/c^2


This best fits the W bosons that have a mass of 80.4 $\frac{GeV}{c^2}$.  It must be a $W^-$ boson as (c) says it decays into an electron and neutrino, so to preserve charge this one should be negative.  But other than that the charge doesn't seem relevent to the problem, as the $W^+$ is the same mass.

----
---

- (b) What is the $\beta$ and $\gamma$.

I can use this relation:<br>
$$E = \gamma mc^2$$
$$\gamma = \frac{E}{mc^2}$$

And for $\beta$ the definition of $\gamma$ can be used:<br>
$$\gamma = \frac{1}{1 - \beta^2}$$<br>
$$1 - \beta^2 = \frac{1}{\gamma}$$<br>
$$\beta^2 = 1 - \frac{1}{\gamma}$$<br>
$$\beta = \sqrt{1 - \frac{1}{\gamma}}$$

In [43]:
E = float(p4_w[0])
gamma = E/m 
print('gamma: %.3f' % (gamma))

beta = np.sqrt(1 - 1 / gamma**2)
print('beta: %.3f ' % (beta))

gamma: 2.462
beta: 0.914 


---
---

- (c) Now boost the particle into its own rest frame. The particle decays 
into an electron and a neutrino. Assume the neutrino travels in the z 
direction in this rest frame. Write down the energy/momentum 4-
vectors for the particle and for the electron and the neutrino (in 
GeV).

In its rest frame, there is no velocity (*hence no momentum*), just rest energy.

So we'll get a new energy momentum 4-vector, with $\vec{p} = \vec{0}$.

$p^{\mu'} = (\frac{E^{'}}{c}, \vec{0})$

But I think I can assume natural units, so $c = 1$

The **length** of the 4-vector is invariant, so the new $p^{u'}p_{\mu '}$ is the same length as $p^\mu p_{\mu '}$:

$$E^2 - \vec{p}^2 = m^2$$
$$200^2 - (30^2 + 100^2 + 150^) = 81.24^2$$
Or the rest energy / invariant mass is 81.24 $\frac{GeV}{c^2}$

So $p^{\mu '} = (81.24 \frac{GeV}{c}, \vec{0})$, which does have the same length and the same invariance as $p^{\mu '}p_{\mu '} = 81.24$ in the new rest frame.


Or $p^{\mu '} = \begin{bmatrix} 81.24 \frac{GeV}{c} \\ 0 \\ 0 \\ 0\end{bmatrix}$, in the rest frame


When it decays into an electron and neutrino, momentum is conserved, so it's $\vec{0}$ in total for them.  The two particles will have equal and opposite momentum, canceling out.

Example 3.3 on page 103 walks through a similar problem, a few ways so I'll use the 4-vector method which seems a bit simpler. We know before and after the collision that energy and momentum are conserved:

$$p_w = p_e + p_\nu$$

We can arrange for $p_e$ and after for $p_\nu$:

$$p_\nu = p_w + p_e$$

Taking the scalar product of these:

$$p_\nu^2 = (p_W + p_e)^2$$
$$m_\nu^2 c^2 = p_w^2 + p_e^2 - 2 p_w p_e$$

In general we get $p^\mu p_\mu$ is the invariant mass:

- $p_\nu^2 = m_\nu^2 c^2 = 0$, due to the tiny mass of the neutrino, roughly $10^{-6} \frac{MeV}{c^2}$
- $p_e^2 = m_e^2 c^2$, this seems like it could also be 0 as it's only $0.511 \frac{MeV}{c^2}$ so about $\frac{1}{4000}$ of a GeV, vs $81.2 \frac{GeV}{c^2}$ for the $W^-$.
- $p_w^2 = m_w^2 c^2$
- $p_w p_e = \frac{E_w}{c}\frac{E_e}{c} = m_w E_e$

$$0 = m_w^2c^2 + 0 - 2 m_w E_e$$
$$E_e = \frac{1}{2}m_wc^2$$

And this seems to be a pretty good answer, all of the energy was initially in the $W^-$ and $m_w c^2$ was its rest energy, so the electron gets half of the energy, leaving the other half for the neutrino.  Momentum has to be equal and opposite for the two particles, and we can ignore their rest mass as they're so tiny and solve for the momentum, at least the magnitude of the momentum 3-vector:

$$p_e = p_w - p_\nu$$
$$(p_e)^2 = (p_w - p_\nu)^2$$
$$p_e^2 = p_w^2 + p_\nu^2 - 2p_w p_\nu$$

We can plug & chug the same values from above:

$$0 = m_w^2c^2 + 0 - 2 m_w E_\nu$$

And in this case, $E_\nu$ is just $|p_\nu|c$ as it's dominated by its momentum, due to having almost zero mass, especially at these scales it picked up a lot of kinetic energy from the decay of the $W^-$.  And since net momentum is 0, we know $|p_\nu| = |p_e|$ and we can replace it and solve to get the values for $e^-$:

$$0 = m^2_wc^2 - 2m_w(|p_e|c)$$
$$|p_e| = \frac{1}{2}m_w c$$
And to reiterate: $|p_e| = |p_\nu|$, the direction is entirely arbitrary, whichever axis I choose I simply have to be consistent and give the same value but opposite sign to the other particle.

p_e = \begin{bmatrix}\frac{m_w c}{2} \\ ? \\ 0 \\ 0\end{bmatrix}

---

This turned out a bit harder than I'd thought, I'm not sure what's left if I use the same value for both momentum and energy... but page 112 comes to the rescue with problem 3.19 that gives the results:

$$E_e = \frac{m_w^2 + m_e^2 - m_\nu^2}{2 m_w}c^2$$

If a particle like $W$ at rest decays into two other particles.

In [61]:
m_e = 511 * 1e-6 # electron in GeV
m_v = 2e-9 # electron anti-neutrino mass, in GeV
m_w = 81.24 # W boson, in GeV
E_w = 81.24

def compute_decay_energy(m_a, m_b, m_c):
  ''' Computes the energy for both output particles b and c, given input particle
  a at rest, from Griffiths, page 112, problem 3.19.a'''

  E_b = (m_a**2 + m_b**2 - m_c**2) / (2.0 * m_a)
  E_c = (m_a**2 + m_c**2 - m_b**2) / (2.0 * m_a)

  return (E_b, E_c)

  
E_e, E_v = compute_decay_energy(m_w, m_e, m_v)
print(f'E of electron: {E_e}')
print(f'E of neutrino: {E_v}')

E of electron: 40.620000001607096
E of neutrino: 40.6199999983929


And for the magnitude of the outgoing momenta of these particles, I was apparently way off.  Part (b) of 3.19 has a formula, that includes a triangle function I never considered.

$$|p_e| = |p_\nu| = \frac{c\sqrt{\lambda(m_w^2, m_e^2, m_\nu^2)}}{2m_w}$$

In [92]:
def triangle_function(x, y, z):
  '''Definitely didn't see this anywhere in the reading!  Seemed to be hand
  waved away in the example problems like getting these values was easy.'''
  return x**2 + y**2 + z**2 - 2*x*y - 2*x*z - 2*y*z

def compute_decay_momenta(m_a, m_b, m_c):
  ''' Computes the momentum of two outbound particles (b, c) from a
  single particle (a) at rest decaying based on their masses.
  source: Griffiths page 112, problem 3.19.b'''

  p = np.sqrt(triangle_function(m_a, m_b, m_c)) / (2 * m_a)

  # same momentum magnitude, direction would be opposite and it's arbitrary
  # the problem did say the neutrino goes along the z-axis, so i'll make
  # the electron go in the negative z.
  return (-p, p)

p_e, p_v = compute_decay_momenta(m_w, m_e, m_v)
print(f'Momentum of electron: {p_e}')
print(f'Momentum of neutrino: {p_v}')

Momentum of electron: -0.4999968549852288
Momentum of neutrino: 0.4999968549852288


In [93]:
# declaring 4-vectors for the two daughter particles
p4_e = np.array([[E_e, 0, 0, p_e]]).T
p4_v = np.array([[E_v, 0, 0, p_v]]).T

p4_w_rest = np.array([[m_w, 0, 0, 0]]).T

# Final sanity check - that the rest p4 vector for the W is the same as the
# combination of the electron and neutrino, so conservation of energy and momentum
# will hold:
p4_w_rest - (p4_e + p4_v)

array([[0.],
       [0.],
       [0.],
       [0.]])

In [94]:
print('W energy/momentum 4-vector (in GeV):')
p4_w_rest

W energy/momentum 4-vector (in GeV):


array([[81.24],
       [ 0.  ],
       [ 0.  ],
       [ 0.  ]])

In [95]:
print('electron energy/momentum 4-vector (in GeV):')
p4_e

electron energy/momentum 4-vector (in GeV):


array([[40.62      ],
       [ 0.        ],
       [ 0.        ],
       [-0.49999685]])

In [96]:
print('neutrino energy/momentum 4-vector (in GeV):')
p4_v

neutrino energy/momentum 4-vector (in GeV):


array([[40.62      ],
       [ 0.        ],
       [ 0.        ],
       [ 0.49999685]])

---

## (d)
- Now consider the lab frame again, and consider the decay of the
particle into an electron and a neutrino. Assume the decay products
could go in any direction. What is the maximum and the minimum
magnitude of the 3-momentum that the electron can have?

The minimum could be approximately 0, the electron comes out at rest with the neutrino siphoning off the kinetic energy, and so the momentum would all be carried with it.  and the maximum would be the full momentum is transferred to the electron instead, and the neutrino pops into existence at rest in the lab frame.  This is how the neutrino was theorized to exist by Pauli, when the data didn't show a sharp spike with the expected energy from $\beta$-decay and instead there was a distribution of energies he figured there might be some other particle that was released, otherwise conservation of energy & possibly momentum & angular momentum would not hold anymore.

So the magnitude would range from 0 to a maximum of the momentum that the $W^-$ had in the lab frame, as momentum is conserved:

In [97]:
# Computing the momentum of the W in the lab frame, that'll be the max momentum
# that any of the new particles could have.
p_max = float(np.sqrt(p4_w[1:].T @ p4_w[1:]))
print(f'Maximum momentum the electron could have: {round(p_max,2)} GeV')

Maximum momentum the electron could have: 182.76 GeV


# \#3
- The $\eta(549)$ meson has spin-0 and is observed to decay to a three-pion final
states by the electromagnetic processes $\eta \rightarrow \pi^0 + \pi^0 + \pi^0$ and $\eta \rightarrow \pi^+ + \pi^- + \pi^0$. Use this information to deduce the parity of the $\eta(549)$, and hence explain why the decays $\eta \rightarrow \pi^0 + \pi^0$ and
$\eta \rightarrow \pi^+ + \pi^-$ have never been observed.

The parity of all pions is (-1) per [wiki](https://en.wikipedia.org/wiki/Pion), and parity is conserved in EM interactions.  To check parity we can just multiply each side of the decay reaction by the parity values, and they have to match for parity to be conserved.  So if there's two particles we'll calculate: $(-1)(-1) = 1$, and the other side of the equation would have to have a $(+1)$ for parity to be conserved

So on the right side of the allowed processes we have for parity:

$$\pi^0 + \pi^0 + \pi^0 \rightarrow \left[(-1)\times(-1)\times(-1)\right] = (-1)$$

And the other allowed process:

$$\pi^+ + \pi^- + \pi^0 \rightarrow \left[(-1)\times(-1)\times(-1)\right] = (-1)$$

That would imply a parity of (-1) for the $\eta(549)$, and checking the never observed processes:

$$\pi^0 + \pi^0 \rightarrow \left[(-1)\times(-1)\right] = (+1)$$

$$\pi^+ + \pi^- \rightarrow \left[\times(-1)\times(-1)\right] = (+1)$$

Both of these outputs give a (+1) parity, and would violate the conservation of parity if $\eta(549)$ had a parity of (-1), hence why they aren't observed.  

Beyond that charge was conserved, baryon number is conserved as they all add up to 0 thanks to the anti-quarks, which also makes the quark flavors balance, I can't find any other law these would violate.

## \#4

### (a)
- Consider the collision of two particles, A and B, which interact and create
n final state particles C1, C2….Cn. For this reaction to occur, there must
be a minimum total energy available, which depends on the final state
particles. This minimum (or threshold) energy corresponds to a final
state of zero kinetic energy in the center-of-momentum frame.
Assuming particle A has total energy E and particle B is at rest, find an
expression for the threshold energy.

Initially in the lab frame the total energy/momentum is:

$p^\mu = (\frac{E_A}{c} + m_Bc, \vec{p}_A)$

This is technically the entire, or total energy/momentum 4-vector, not just for one particular object but there's enough subscripts & superscripts floating I don't want to add more to it.

The energy of particle A, both rest and kinetic, combined with the rest energy of the target particle B.  And all of the momentum is from particle A, along the x-axis though it's abitrary and I'll switch to the CM frame soon enough, where total momentum will be zero, both before & after the collision.

In the CM frame the $p^{\mu '}$ is even easier:

$p^{\mu '} = ((m_1 + m_2 + m_3 + \dots + m_n)c, \vec{0})$

Or we can just use $M$ for all the masses of these particles that pop out:

$p^{\mu '} = (Mc, \vec{0})$

So that makes this simple invariant scaler product simple:

$$p^\mu p_\mu = (Invariant) = M^2c^2$$

But that's post-collision in the CM frame.  Maybe some useful info can be pulled along from the lab pre-collision with the same invariant, then conservation of energy or momentum in the CM frame can solve for the threshold energy:

$$p^\mu p_\mu = (\frac{E_A}{c} + m_Bc)^2 - \vec{p}^2_A$$
$$(Invariant) = \frac{E_A^2}{c^2} + m^2_Bc^2 + 2m_BE_A - \vec{p}_A^2$$
$$(Invariant) = (\frac{E_A^2}{c^2} - \vec{p}_A^2) + m^2_Bc^2 + 2m_BE_A$$
Now that those terms are collected in the first group I can use:

$$E^2_A - \vec{p}^2c^2 = m_A^2c^4$$
Or:
$$\left(\frac{E^2_A}{c^2} - \vec{p}^2\right) = m_A^2c^2$$
And:

$$(Invariant) = (m_A^2c^2) + m^2_Bc^2 + 2m_BE_A$$
$$(Invariant) = (m_A^2 + m^2_B)c^2 + 2m_BE_A$$
Using the earlier invariant from the lab frame:
$$M^2c^2 = (m_A^2 + m^2_B)c^2 + 2m_BE_A$$<br>
$$2m_BE_A = M^2 - (m_A^2 + m^2_B)c^2$$<br>

---

$$E_A = \frac{M^2 - (m_A^2 + m^2_B)}{2m_B}$$

---

Energy and momentum are conserved across the collisions, and the invariants can be moved between the lab & CM frame.

### (b)
- Use your answer from part (a) to find the threshold energies for the
following reactions. In each case, the proton is the at-rest target
particle:

  - $\pi^- + p \rightarrow K^0 + \Sigma^0$
  - $p + p \rightarrow p + p + \pi^0$
  - $\pi^- + p \rightarrow p + \bar{p} + n$

These should just take the masses as input, and that'll give the energy required:

In [90]:
# all masses in MeV/c^2
m_proton = 938.3
m_pion = 139.6
m_kaon = 497.7
m_sigma = 1192.5
m_neutron = 939.6
m_pion_0 = 135


def calc_threshold_energy(incoming, created):
  M = sum(created)
  E_A = (M**2 - (incoming[0]**2 + incoming[1]**2)) / (2 * incoming[1])
  return round(E_A, 3)


exp_in = [m_pion, m_proton]
exp_out = [m_kaon, m_sigma]

E_1 = calc_threshold_energy(exp_in, exp_out)
print(f'Threshold energy for first reaction: {E_1}')

exp_in = [m_proton, m_proton]
exp_out = [m_proton, m_proton, m_pion_0]
E_2 = calc_threshold_energy(exp_in, exp_out)
print(f'Threshold energy for second reaction: {E_2}')

exp_in = [m_pion, m_proton]
exp_out = [m_proton, m_proton, m_neutron]
E_3 = calc_threshold_energy(exp_in, exp_out)
print(f'Threshold energy for third reaction: {E_3}')

Threshold energy for first reaction: 1042.78
Threshold energy for second reaction: 1218.012
Threshold energy for third reaction: 3746.716


---
---

## \#4

Following the discussion of the triangle symmetry group in Griffiths (Ch 4.1,
pp 117-118), work out the symmetry group of the square.

- (a) How many elements does it have? [Hint: Draw a diagram of the axes of
symmetry. Each symmetry group element for the square must
symmetrically preserve the orientation of the original square.]

There are 8 elements:
1. I, The identity mapping
1. R, $90^o$ rotation, counter-clockwise (CCW)
1. R2, $180^o$ rotation, counter-clockwise (CCW)
1. R3, $270^o$ rotation, counter-clockwise (CCW)
1. Tx, Reflection across the x-axis
1. Ty, Reflection across the y-axis
1. Tac, Reflection across the diagonals, labeled AC
1. Tbd, Reflection across the other diagonal, labeled BD

- (b) Determine if the square symmetry group is Abelian or non-Abelian.

Proofs are a little above my pay grade, but maybe a counter example should do:

If we do $T_xR3 \rightarrow T_{BD}$<br>
But the inverse $R3T_x \rightarrow T_{AC}$:

So not all group operations are commutative, and the square symmetry group is not abelian.
